modules/apps/portal/portal-upgrade-impl/src/main/java/com/liferay/portal/upgrade/internal/report/UpgradeReport.java

```diff
@@ -45,10 +67,111 @@ public void addWarningMessage(String loggerName, String message) {
 		warningMessages.add(message);
 	}
 
+	public void generateReport() {
+		File file = new File(PropsValues.LIFERAY_HOME, "upgrade_report.info");
+
+		try {
+			FileUtil.write(file, _getLiferayVersions());
+
+			FileUtil.write(file, _getDialectInfo(), false, true);
+
+			FileUtil.write(file, _getProperties(), false, true);
+		}
+		catch (IOException ioException) {
+		}
+	}
+
+	private String _getDialectInfo() {
+		StringBuffer sb = new StringBuffer(7);
+
+		DB db = DBManagerUtil.getDB();
+
+		sb.append("Using ");
+		sb.append(db.getDBType());
+		sb.append(" version ");
+		sb.append(db.getMajorVersion());
+		sb.append(StringPool.PERIOD);
+		sb.append(db.getMinorVersion());
+		sb.append(StringPool.NEW_LINE);
+
+		return sb.toString();
+	}
+
+	private int _getInitialBuildNumber() {
+		try (Connection connection = DataAccess.getConnection();
+			PreparedStatement preparedStatement = connection.prepareStatement(
+				"select buildNumber from Release_ where releaseId = " +
+					ReleaseConstants.DEFAULT_ID)) {
+
+			ResultSet resultSet = preparedStatement.executeQuery();
+
+			if (resultSet.next()) {
+				return resultSet.getInt("buildNumber");
+			}
+		}
+		catch (Exception exception) {
+		}
+
+		return -1;
+	}
+
+	private String _getLiferayVersions() {
+		StringBuffer sb = new StringBuffer(6);
+
+		if (_initialBuildNumber != -1) {
+			sb.append("Initial Version of Liferay: ");
+			sb.append(_initialBuildNumber);
+			sb.append(StringPool.NEW_LINE);
+		}
+		else {
+			sb.append("Unable to determine initial version of Liferay.");
+			sb.append(StringPool.NEW_LINE);
+		}
+
+		sb.append("Final Version of Liferay: ");
+		sb.append(ReleaseInfo.getBuildNumber());
+		sb.append(StringPool.NEW_LINE);
+
+		return sb.toString();
+	}
+
+	private String _getProperties() {
+		StringBuffer sb = new StringBuffer(10);
+
+		String dlStore = PropsValues.DL_STORE_IMPL;
+
+		sb.append(PropsKeys.DL_STORE_IMPL + StringPool.EQUAL + dlStore);
+
+		sb.append(StringPool.NEW_LINE);
+
+		if (dlStore.contains("FileSystemStore")) {
+			sb.append(
+				"Please check your OSGi configuration files to ensure " +
+					"rootDir has been set properly");
+
+			sb.append(StringPool.NEW_LINE);
+		}
+
+		sb.append(
+			"locales.enabled=" + Arrays.toString(PropsValues.LOCALES_ENABLED));
+		sb.append(StringPool.NEW_LINE);
+		sb.append("locales=" + Arrays.toString(PropsValues.LOCALES));
+		sb.append(StringPool.NEW_LINE);
+		sb.append("liferay.home=" + PropsValues.LIFERAY_HOME);
+		sb.append(StringPool.NEW_LINE);
+
+		return sb.toString();
+	}
+
+	private void _setInitialBuildNumber() {
+		_initialBuildNumber = _getInitialBuildNumber();
```

Comment:
@SamZiemer Since _getInitialBuildNumber is not called in other places, move its logic here and delete the method